### ISSUE TITLE
Automatic Difficulty Adjustment for Each User in XMRig Proxy "Reduce worker diff"

### ISSUE THREAD

**Issue by vacSQL** [xmrig/xmrig-proxy#572]
I am currently using XMRig Proxy for solo mining.
Could you please consider adding an option in XMRig Proxy to automatically adjust the difficulty for each user based on their mining hashrate? This feature would optimize mining efficiency by ensuring that miners with varying hashrates receive tasks that match their computational capabilities.
Having this dynamic difficulty adjustment feature would simplify the setup process and improve overall mining performance. It would allow miners to contribute more effectively without manual adjustment of difficulty settings.
Thank you for your consideration

**Comment by XY-M-XY** [comment]
Finally, someone brought it up. Yes, the difficulty needs to be set automatically for each miner instead of being fixed. If the difficulty is too high, once a small miner is accidentally disconnected, all previous work will be in vain.

**Comment by SChernykh** [comment]
> once a small miner is accidentally disconnected, all previous work will be in vain

No, it will not be. Mining is a probabilistic process, it has no "memory" of previous work. Each hash has a chance to find a block, or  a share with big enough difficulty. Please educate yourself: https://en.wikipedia.org/wiki/Memorylessness

**Comment by XY-M-XY** [comment]
Sorry, I didn't express myself clearly. The "waste" I mentioned refers to the proxy. If the difficulty is high, the proxy will not be able to calculate the hash value of the small miner and it will always be 0. This will not happen if the difficulty is dynamic.
